src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,262 @@
+package parser;
+
+import static ui.Ui.echo;
+import static ui.Ui.filterList;
+import static ui.Ui.line;
+import static ui.Ui.listOut;
+import static ui.Ui.stringFilterList;
+import static ui.Ui.stringListOut;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import data.task.Task;
+import storage.Storage;
+import tasklist.TaskList;
+import ui.Ui;
+
+public class Parser {
+    Parser() {
+        // empty constructor
+    }
+
+    /**
+     * accepts user inputs
+     * @param tl list of current tasks
+     * @param stores deals with location of storage file
+     */
+    public static void accept(TaskList tl, Storage stores) {
+        Scanner scanner = new Scanner(System.in);
+        while (scanner.hasNextLine()) {
+            String output = scanner.nextLine();
+            // special 1 word cases
+            // bye exits while
+            if (output.equals("bye")) {
+                echo("Ciao!");
+                break;
+            } else if (output.equals("list")) {
+                listOut(tl.getStorage());
+            } else {
+                Scanner multiWord = new Scanner(output);
+                // can't use enums here, firstWord can be literally anything
+                String firstWord = multiWord.next();
+                // wow intelliJ is a better programmer than i'll ever be
+                switch (firstWord) {
+                case "done": {
+                    String index = "0";
+                    try {
+                        if (multiWord.hasNext()) {
+                            index = multiWord.next();
+                        } else {
+                            throw Ui.DukeException.empty("done");
+                        }
+                    } catch (Ui.DukeException e) {
+                        echo(e.getMessage());
+                    }
+                    int intIndex = Integer.parseInt(index);
+                    try {
+                        if (intIndex <= tl.getStorage().size() && intIndex > 0) {
+                            tl.markComplete(intIndex - 1);
+                            // save to save file
+                            stores.save(tl.getStorage());
+                        } else {
+                            throw Ui.DukeException.outOfBounds(intIndex);
+                        }
+                    } catch (Ui.DukeException e) {
+                        echo(e.getMessage());
```

Comment:
Code here is not readable nor intuitive, can consider handling the parsing of input in DoneCommand and storage of task in Storage